Userland/Services/WebContent/WebDriverConnection.cpp

```diff
@@ -1951,12 +1955,14 @@ ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::wait_for_navigation_to
     }();
 
     // 6. Wait for the current browsing context’s document readiness state to reach readiness target,
-    // FIXME: or for the session page load timeout to pass, whichever occurs sooner.
+    //    or for the session page load timeout to pass, whichever occurs sooner.
     Web::Platform::EventLoopPlugin::the().spin_until([&]() {
-        return m_page_client->page().top_level_browsing_context().active_document()->readiness() == readiness_target;
+        return page_load_timeout_fired || page_load_timeout_fired || m_page_client->page().top_level_browsing_context().active_document()->readiness() == readiness_target;
```

Comment:
SerenityOS's CI rightfully remarks:

```
/home/runner/work/serenity/serenity/Userland/Services/WebContent/WebDriverConnection.cpp:1954:40: error: logical 'or' of equal expressions [-Werror=logical-op]
 1954 |         return page_load_timeout_fired || page_load_timeout_fired || m_page_client->page().top_level_browsing_context().active_document()->readiness() == readiness_target;
      |                ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
```

@trflynn89 has since fixed this, but do you know why your CI didn't see this?

Reply:
```
(venv) andrew@Andrews-MacBook-Pro-2:~/Source/ladybird-browser$ git grep logical-op
(venv) andrew@Andrews-MacBook-Pro-2:~/Source/ladybird-browser$ git -C ../serenity grep logical-op
Meta/CMake/serenity_compile_options.cmake:add_compile_options(-Wlogical-op)
```

Looks like we should scan through all the compile options that serenity uses and add them to our flags. Also that flag not appearing in common compile options is odd.

Reply:
Serenity has `-Wlogical-op` enabled, Lagom / Ladybird do not:
https://github.com/SerenityOS/serenity/blob/master/Meta/CMake/serenity_compile_options.cmake#L11

As for why that is, no idea - probably an oversight years ago

Reply:
"has since fixed this": In https://github.com/LadybirdBrowser/ladybird/pull/1389

@ADKaster Sounds like you might want to build with -Wlogical-op!

Reply:
> "has since fixed this": In #1389

Actually, in #1644.